### ISSUE TITLE
Merge weaveDNS executable into router executable

### ISSUE THREAD

**Issue by awh** [weaveworks/weave#843]
* Merge weavedns main into weaver main:
   - Prefix all dns options with 'dns-'
   - Merge signal handlers
   - Merge http handlers (at least make sure they don't trip over each other)
   - Merge status output
   - Merge log configuration
   - Merge docker event listeners (at least make sure they don't trip over each other)
* Ensure log entries from router and dns can be distinguished
* Update `weave` script:
   - Configure an ethwe, in the normal way, on weave container, if `--dns <cidr>` is specified
   - Don't launch a separate container when `--dns` is specified; pass a flag to the weave container start instead that enables dns.

Depends on #842.

**Comment by tomwilkie** [comment]
This has been superseded by #826
